app/models/dfe_sign_in_user.rb

```diff
@@ -63,6 +63,10 @@ def user
     @user ||= user_by_uid || user_by_email
   end
 
+  def system_admin?
+    user.present? && user.system_admin?
```

Comment:
could this be `user&.system_admin?` as you've done for the `OtpSignInUser`?

Reply:
I'm not sure about this because I expect a method called `system_admin?` to return a boolean and it won't if `user` is `nil`. `!!user&.system_admin?` works but you don't see this much in Ruby.

Reply:
good point, it's reasonable to expect this method to return a boolean. doesn't the same logic apply to the OtpSignInUser?

fwiw i don't object to `!!user&.system_admin?` or `user&.system_admin? || false`